From b220c377941d0b1271cf525a8d06bb8e48196d2b Mon Sep 17 00:00:00 2001
From: Joey Hess <joey@kitenet.net>
Date: Thu, 28 Feb 2013 23:29:04 -0400
Subject: [PATCH] disable TH

---
 aeson.cabal |    1 -
 1 file changed, 1 deletion(-)

diff --git a/aeson.cabal b/aeson.cabal
index 242aa67..275aa49 100644
--- a/aeson.cabal
+++ b/aeson.cabal
@@ -99,7 +99,6 @@ library
     Data.Aeson.Generic
     Data.Aeson.Parser
     Data.Aeson.Types
-    Data.Aeson.TH
 
   other-modules:
     Data.Aeson.Functions
-- 
1.7.10.4

