From 55f424de9946c4d1d89837bb18698437aecfcfa4 Mon Sep 17 00:00:00 2001
From: Joey Hess <joey@kitenet.net>
Date: Thu, 28 Feb 2013 23:29:16 -0400
Subject: [PATCH] allow building with unreleased ghc

---
 async.cabal |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/async.cabal b/async.cabal
index 8e47d9d..ff317c7 100644
--- a/async.cabal
+++ b/async.cabal
@@ -70,7 +70,7 @@ source-repository head
 
 library
     exposed-modules:     Control.Concurrent.Async
-    build-depends:       base >= 4.3 && < 4.7, stm >= 2.2 && < 2.5
+    build-depends:       base >= 4.3 && < 4.8, stm >= 2.2 && < 2.5
 
 test-suite test-async
     type:       exitcode-stdio-1.0
-- 
1.7.10.4

