From d456247000ab839a1d32749717f4f8f92e37dbba Mon Sep 17 00:00:00 2001
From: Joey Hess <joey@kitenet.net>
Date: Tue, 7 May 2013 17:45:45 -0400
Subject: [PATCH] fix cross build

---
 cipher-aes.cabal |    5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/cipher-aes.cabal b/cipher-aes.cabal
index 02ddfd0..eb916e3 100644
--- a/cipher-aes.cabal
+++ b/cipher-aes.cabal
@@ -31,16 +31,13 @@ Extra-Source-Files:  Tests/*.hs
 
 Library
   Build-Depends:     base >= 4 && < 5
-                   , bytestring
+                   , bytestring >= 0.10.3.0
   Exposed-modules:   Crypto.Cipher.AES
   ghc-options:       -Wall
   C-sources:         cbits/aes_generic.c
                      cbits/aes.c
                      cbits/gf.c
                      cbits/cpu.c
-  if os(linux) && (arch(i386) || arch(x86_64))
-    CC-options:      -mssse3 -maes -mpclmul -DWITH_AESNI
-    C-sources:       cbits/aes_x86ni.c
 
 Test-Suite test-cipher-aes
   type:              exitcode-stdio-1.0
-- 
1.7.10.4

