From 6cd7b7d3a8e38ada9b1e3989770525c63f9f1d7d Mon Sep 17 00:00:00 2001
From: androidbuilder <androidbuilder@example.com>
Date: Fri, 18 Oct 2013 23:10:16 +0000
Subject: [PATCH] cross build

---
 distributive.cabal |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/distributive.cabal b/distributive.cabal
index 5d4ac69..82d7593 100644
--- a/distributive.cabal
+++ b/distributive.cabal
@@ -12,7 +12,7 @@ bug-reports:   http://github.com/ekmett/distributive/issues
 copyright:     Copyright (C) 2011-2013 Edward A. Kmett
 synopsis:      Haskell 98 Distributive functors -- Dual to Traversable
 description:   Haskell 98 Distributive functors -- Dual to Traversable
-build-type:    Custom
+build-type:    Simple
 extra-source-files:
   .ghci
   .travis.yml
-- 
1.7.10.4

