From 35c6718205e9d7f5e5fc44578ea6a9971beac151 Mon Sep 17 00:00:00 2001
From: foo <foo@bar>
Date: Sat, 21 Sep 2013 23:32:18 +0000
Subject: [PATCH] cross build

---
 entropy.cabal |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/entropy.cabal b/entropy.cabal
index 45e4705..17553d8 100644
--- a/entropy.cabal
+++ b/entropy.cabal
@@ -14,7 +14,7 @@ category:       Data, Cryptography
 homepage:       https://github.com/TomMD/entropy
 bug-reports:    https://github.com/TomMD/entropy/issues
 stability:      stable
-build-type:     Custom
+build-type:     Simple
 cabal-version:  >= 1.6
 tested-with:    GHC == 6.12.1
 data-files:
-- 
1.7.10.4

