From 8e942c1f661b30e5477607b78528634e6d345ae8 Mon Sep 17 00:00:00 2001
From: androidbuilder <androidbuilder@example.com>
Date: Thu, 2 Jul 2015 21:16:15 +0000
Subject: [PATCH] cross build

---
 entropy.cabal | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/entropy.cabal b/entropy.cabal
index e4fb436..e26896c 100644
--- a/entropy.cabal
+++ b/entropy.cabal
@@ -14,10 +14,7 @@ category:       Data, Cryptography
 homepage:       https://github.com/TomMD/entropy
 bug-reports:    https://github.com/TomMD/entropy/issues
 stability:      stable
--- build-type:  Simple
--- ^^ Used for HaLVM
-build-type:        Custom
--- ^^ Test for RDRAND support using 'ghc'
+build-type:  Simple
 cabal-version:  >=1.10
 tested-with:    GHC == 7.8.2
 -- data-files:
-- 
2.1.4
