From fdbd29ce6e8ff11f721f9e74cac1f4ca14e6773d Mon Sep 17 00:00:00 2001
From: foo <foo@bar>
Date: Sun, 22 Sep 2013 07:06:33 +0000
Subject: [PATCH] export TH symbols

---
 Data/FileEmbed.hs |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/Data/FileEmbed.hs b/Data/FileEmbed.hs
index c17f082..6654f60 100644
--- a/Data/FileEmbed.hs
+++ b/Data/FileEmbed.hs
@@ -26,6 +26,8 @@ module Data.FileEmbed
 #endif
     , inject
     , injectFile
+    -- used by TH (pointlessly)
+    , stringToBs
     ) where
 
 import Language.Haskell.TH.Syntax
-- 
1.7.10.4

