From 9819f4b387679c889f1259f9fd969513aa2efcf2 Mon Sep 17 00:00:00 2001
From: foo <foo@bar>
Date: Sun, 22 Sep 2013 03:51:06 +0000
Subject: [PATCH] export TH splice stuff

---
 Text/Hamlet.hs |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/Text/Hamlet.hs b/Text/Hamlet.hs
index 6568d6c..687dec4 100644
--- a/Text/Hamlet.hs
+++ b/Text/Hamlet.hs
@@ -40,6 +40,11 @@ module Text.Hamlet
     , ihamletRules
     , htmlRules
     , CloseStyle (..)
+    -- referred to by TH splices
+    , asHtmlUrl
+    , maybeH
+    , condH
+    , attrsToHtml
     ) where
 
 import Text.Shakespeare.Base
-- 
1.7.10.4

