From 643b3c9fd95967c5911107f46498cd851e68f97d Mon Sep 17 00:00:00 2001
From: Joey Hess <joey@kitenet.net>
Date: Tue, 7 May 2013 18:26:33 -0400
Subject: [PATCH] fix build

---
 hS3.cabal |    3 ---
 1 file changed, 3 deletions(-)

diff --git a/hS3.cabal b/hS3.cabal
index 35f7496..e04bf65 100644
--- a/hS3.cabal
+++ b/hS3.cabal
@@ -44,6 +44,3 @@ Library
         Network.AWS.AWSConnection,
         Network.AWS.Authentication,
         Network.AWS.ArrowUtils
-
-Executable     hs3
-  main-is:        hS3.hs
-- 
1.7.10.4

