From 32d0741c64e6bd280e46f7c452db9462fbac05f9 Mon Sep 17 00:00:00 2001
From: Joey Hess <joey@kitenet.net>
Date: Tue, 7 May 2013 18:21:04 -0400
Subject: [PATCH] fix build

---
 HTTP.cabal |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/HTTP.cabal b/HTTP.cabal
index 76cb5d6..edddf26 100644
--- a/HTTP.cabal
+++ b/HTTP.cabal
@@ -85,12 +85,12 @@ Library
                  Network.HTTP.Utils
                  Paths_HTTP
   GHC-options: -fwarn-missing-signatures -Wall
-  Build-depends: base >= 2 && < 4.7, network < 2.5, parsec
+  Build-depends: base >= 2 && < 4.8, network < 2.5, parsec
   Extensions: FlexibleInstances
   if flag(old-base)
     Build-depends: base < 3
   else
-    Build-depends: base >= 3, array, old-time, bytestring
+    Build-depends: base >= 3, array, old-time, bytestring (>= 0.10.3.0)
 
   if flag(mtl1)
     Build-depends: mtl >= 1.1 && < 1.2
-- 
1.7.10.4

