From d4c861dbdee34cb2434085b9ece62c416d4cad79 Mon Sep 17 00:00:00 2001
From: androidbuilder <androidbuilder@example.com>
Date: Sat, 8 Feb 2014 17:19:37 +0000
Subject: [PATCH] text dependency

---
 libxml-sax.cabal |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libxml-sax.cabal b/libxml-sax.cabal
index 60dba81..d6883bd 100644
--- a/libxml-sax.cabal
+++ b/libxml-sax.cabal
@@ -35,7 +35,7 @@ library
   build-depends:
       base >= 4.1 && < 5.0
     , bytestring >= 0.9
-    , text >= 0.7 && < 0.12
+    , text
     , xml-types >= 0.3 && < 0.4
 
   exposed-modules:
-- 
1.7.10.4

