From 8c4220e4dd48ad197aa0ad49214e6e7bd768044e Mon Sep 17 00:00:00 2001
From: Joey Hess <joey@kitenet.net>
Date: Thu, 28 Feb 2013 23:28:57 -0400
Subject: [PATCH] fix build (not Android specific)

---
 src/System/Cmd/Utils.hs |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/System/Cmd/Utils.hs b/src/System/Cmd/Utils.hs
index a9fa46f..6c6aba2 100644
--- a/src/System/Cmd/Utils.hs
+++ b/src/System/Cmd/Utils.hs
@@ -325,7 +325,7 @@ forceSuccess (PipeHandle pid fp args funcname) =
                 Just (Exited (ExitSuccess)) -> return ()
                 Just (Exited (ExitFailure fc)) ->
                     cmdfailed funcname fp args fc
-                Just (Terminated sig) ->
+                Just (Terminated sig _) ->
                     warnfail fp args $ "Terminated by signal " ++ show sig
                 Just (Stopped sig) ->
                     warnfail fp args $ "Stopped by signal " ++ show sig
@@ -351,7 +351,7 @@ safeSystem command args =
        case ec of
             Exited ExitSuccess -> return ()
             Exited (ExitFailure fc) -> cmdfailed "safeSystem" command args fc
-            Terminated s -> cmdsignalled "safeSystem" command args s
+            Terminated s _ -> cmdsignalled "safeSystem" command args s
             Stopped s -> cmdsignalled "safeSystem" command args s
 #endif
 
-- 
1.7.10.4

