From 3dde0175096903207c9774d8f6bba9b81ab6c2f9 Mon Sep 17 00:00:00 2001
From: Joey Hess <joey@kitenet.net>
Date: Thu, 28 Feb 2013 23:31:45 -0400
Subject: [PATCH] build with newer ghc

---
 monad-control.cabal |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/monad-control.cabal b/monad-control.cabal
index 2e3eb46..b12ffaf 100644
--- a/monad-control.cabal
+++ b/monad-control.cabal
@@ -56,7 +56,7 @@ Library
 
   Exposed-modules: Control.Monad.Trans.Control
 
-  Build-depends: base                 >= 3     && < 4.7
+  Build-depends: base                 >= 3     && < 4.8
                , base-unicode-symbols >= 0.1.1 && < 0.3
                , transformers         >= 0.2   && < 0.4
                , transformers-base    >= 0.4.1 && < 0.5
-- 
1.7.10.4

