From 8f124aad6d04abba5729af21ba3b50944f165d4b Mon Sep 17 00:00:00 2001
From: androidbuilder <androidbuilder@example.com>
Date: Sat, 8 Feb 2014 17:20:41 +0000
Subject: [PATCH] text dependency

---
 network-protocol-xmpp.cabal |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/network-protocol-xmpp.cabal b/network-protocol-xmpp.cabal
index 2500075..d709a15 100644
--- a/network-protocol-xmpp.cabal
+++ b/network-protocol-xmpp.cabal
@@ -36,7 +36,7 @@ library
     , libxml-sax >= 0.7 && < 0.8
     , monads-tf >= 0.1 && < 0.2
     , network >= 2.2
-    , text >= 0.10 && < 0.12
+    , text
     , transformers >= 0.2
     , xml-types >= 0.3 && < 0.4
 
-- 
1.7.10.4

