From cf110acc7f5863bb80ba835a009a7f59d3453239 Mon Sep 17 00:00:00 2001
From: dummy <dummy@example.com>
Date: Thu, 2 Jul 2015 20:19:14 +0000
Subject: [PATCH] fix build

---
 Network/BSD.hsc | 1 -
 1 file changed, 1 deletion(-)

diff --git a/Network/BSD.hsc b/Network/BSD.hsc
index e11ac71..039d0f1 100644
--- a/Network/BSD.hsc
+++ b/Network/BSD.hsc
@@ -396,7 +396,6 @@ instance Storable NetworkEntry where
    poke _p = error "Storable.poke(BSD.NetEntry) not implemented"
 
 
-#if !defined(cygwin32_HOST_OS) && !defined(mingw32_HOST_OS) && !defined(_WIN32)
 getNetworkByName :: NetworkName -> IO NetworkEntry
 getNetworkByName name = withLock $ do
  withCString name $ \ name_cstr -> do
-- 
2.1.4

