From 5cb5c3dabb213f809b8328b0b4049f7c754e9c77 Mon Sep 17 00:00:00 2001
From: Joey Hess <joey@kitenet.net>
Date: Thu, 28 Feb 2013 23:34:32 -0400
Subject: [PATCH] disable i386 opt stuff to allow cross-compilation

---
 primitive.cabal |    3 ---
 1 file changed, 3 deletions(-)

diff --git a/primitive.cabal b/primitive.cabal
index 8c4328a..9a6093f 100644
--- a/primitive.cabal
+++ b/primitive.cabal
@@ -51,7 +51,4 @@ Library
   includes: primitive-memops.h
   c-sources: cbits/primitive-memops.c
   cc-options: -O3 -ftree-vectorize -fomit-frame-pointer
-  if arch(i386) || arch(x86_64) {
-    cc-options: -msse2
-  }
 
-- 
1.7.10.4

