From ff72a2c501358b3cd71e15c6275ee03670748f3d Mon Sep 17 00:00:00 2001
From: Joey Hess <joey@kitenet.net>
Date: Tue, 7 May 2013 17:52:50 -0400
Subject: [PATCH] avoid configure script

---
 process.cabal |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/process.cabal b/process.cabal
index 8c76b08..2930fdc 100644
--- a/process.cabal
+++ b/process.cabal
@@ -14,7 +14,7 @@ extra-source-files:
 extra-tmp-files:
     config.log config.status autom4te.cache
     include/HsProcessConfig.h
-build-type:    Configure
+build-type:    Simple
 cabal-version: >=1.6
 
 source-repository head
-- 
1.7.10.4

