From 0b0d4250cfce44b1a03b50458b4122370ab349ce Mon Sep 17 00:00:00 2001
From: foo <foo@bar>
Date: Sat, 21 Sep 2013 21:50:51 +0000
Subject: [PATCH] fix build with new ghc

---
 System/Process/Internals.hs |    1 +
 1 file changed, 1 insertion(+)

diff --git a/System/Process/Internals.hs b/System/Process/Internals.hs
index a73c6fc..6676a72 100644
--- a/System/Process/Internals.hs
+++ b/System/Process/Internals.hs
@@ -61,6 +61,7 @@ import Control.Concurrent
 import Control.Exception
 import Foreign.C
 import Foreign
+import System.IO.Unsafe
 
 # ifdef __GLASGOW_HASKELL__
 
-- 
1.7.10.4

