From 40182bfb77ba16beab0da95b664d2c052d5fcad6 Mon Sep 17 00:00:00 2001
From: foo <foo@bar>
Date: Sun, 22 Sep 2013 04:53:30 +0000
Subject: [PATCH] TH exports

---
 Text/Julius.hs |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/Text/Julius.hs b/Text/Julius.hs
index 3a9f83e..2b98f30 100644
--- a/Text/Julius.hs
+++ b/Text/Julius.hs
@@ -40,6 +40,8 @@ module Text.Julius
     , javascriptSettings
       -- ** internal
     , juliusUsedIdentifiers
+      -- used by TH
+    , asJavascriptUrl
     ) where
 
 import Language.Haskell.TH.Quote (QuasiQuoter (..))
-- 
1.7.10.4

