From 2feaef797641587a3da83753ee17d20e712c79cf Mon Sep 17 00:00:00 2001
From: Joey Hess <joey@kitenet.net>
Date: Thu, 28 Feb 2013 23:36:30 -0400
Subject: [PATCH] modify to build with unreleased ghc

---
 split.cabal |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/split.cabal b/split.cabal
index 2183c3e..29b9b32 100644
--- a/split.cabal
+++ b/split.cabal
@@ -51,7 +51,7 @@ Source-repository head
 
 Library
   ghc-options:       -Wall
-  build-depends:     base <4.7
+  build-depends:     base <4.8
   exposed-modules:   Data.List.Split, Data.List.Split.Internals
   default-language:  Haskell2010
   Hs-source-dirs:    src
-- 
1.7.10.4

