From c40fe2c484096c5de4cac8ca14a0ca5d892999f7 Mon Sep 17 00:00:00 2001
From: Joey Hess <joey@kitenet.net>
Date: Thu, 28 Feb 2013 23:36:43 -0400
Subject: [PATCH] hack for cross-compiling

---
 syb.cabal |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/syb.cabal b/syb.cabal
index 0aee93d..0a645c6 100644
--- a/syb.cabal
+++ b/syb.cabal
@@ -17,7 +17,7 @@ description:
 
 category:               Generics
 stability:              provisional
-build-type:             Custom
+build-type:             Simple
 cabal-version:          >= 1.6
 
 extra-source-files:     tests/*.hs,
-- 
1.7.10.4

