From 0f6fe7d649993223ec8987b1aee652850e9d10f3 Mon Sep 17 00:00:00 2001
From: Joey Hess <joey@kitenet.net>
Date: Tue, 7 May 2013 16:08:28 -0400
Subject: [PATCH] hacks for android

---
 cbits/conv.c    | 2 +-
 unix-time.cabal | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/cbits/conv.c b/cbits/conv.c
index 3b6a129..895e6b7 100644
--- a/cbits/conv.c
+++ b/cbits/conv.c
@@ -51,7 +51,7 @@ time_t c_parse_unix_time_gmt(char *fmt, char *src) {
 #else
     strptime(src, fmt, &dst);
 #endif
-    return timegm(&dst);
+    return NULL; /* timegm(&dst); */
 }
 
 void c_format_unix_time(char *fmt, time_t src, char* dst, int siz) {
diff --git a/unix-time.cabal b/unix-time.cabal
index a905d63..2de06c6 100644
--- a/unix-time.cabal
+++ b/unix-time.cabal
@@ -8,7 +8,7 @@ Synopsis:               Unix time parser/formatter and utilities
 Description:            Fast parser\/formatter\/utilities for Unix time
 Category:               Data
 Cabal-Version:          >= 1.10
-Build-Type:             Configure
+Build-Type:             Simple
 Extra-Source-Files:     cbits/conv.c cbits/config.h.in configure configure.ac
 Extra-Tmp-Files:        config.log config.status autom4te.cache cbits/config.h
 
@@ -21,7 +21,7 @@ Library
                         Data.UnixTime.Types
                         Data.UnixTime.Sys
   Build-Depends:        base >= 4 && < 5
-                      , bytestring
+                      , bytestring (>= 0.10.3.0)
                       , old-time
   C-Sources:            cbits/conv.c
 
-- 
1.8.2.rc3

