From 04a1230cf4d740d37ab427165eef4b4db2a3898f Mon Sep 17 00:00:00 2001
From: dummy <dummy@example.com>
Date: Fri, 3 Jul 2015 02:20:42 +0000
Subject: [PATCH] build without v1 uuid which needs network-info

---
 Data/UUID/Util.hs | 11 -----------
 uuid.cabal        |  2 --
 2 files changed, 13 deletions(-)

diff --git a/Data/UUID/Util.hs b/Data/UUID/Util.hs
index 8817f51..0d43b01 100644
--- a/Data/UUID/Util.hs
+++ b/Data/UUID/Util.hs
@@ -3,7 +3,6 @@ module Data.UUID.Util (
     UnpackedUUID(..)
   , unpack, pack
   , version
-  , extractMac
   , extractTime
   , setTime
   ) where
@@ -13,7 +12,6 @@ import Data.Word
 import Data.Word.Util
 import Data.Bits
 import Data.UUID.Types.Internal
-import Network.Info
 import Data.Int (Int64)
 
 version :: UUID -> Int
@@ -42,12 +40,3 @@ extractTime uuid =
 
 timeAndVersionToTime :: Word16 -> Word16
 timeAndVersionToTime tv = tv .&. 0x0FFF
-
-extractMac :: UUID -> Maybe MAC
-extractMac uuid =
-  if version uuid == 1
-  then Just $
-       MAC (node_0 unpacked) (node_1 unpacked) (node_2 unpacked) (node_3 unpacked) (node_4 unpacked) (node_5 unpacked)
-  else Nothing
-  where
-    unpacked = unpack uuid
diff --git a/uuid.cabal b/uuid.cabal
index 2fa548b..9d86fd2 100644
--- a/uuid.cabal
+++ b/uuid.cabal
@@ -30,7 +30,6 @@ Library
                 binary >= 0.4 && < 0.8,
                 bytestring >= 0.9 && < 0.11,
                 cryptohash >= 0.7 && < 0.12,
-                network-info == 0.2.*,
                 random >= 1.0.1 && < 1.2,
                 time >= 1.1 && < 1.6,
                 uuid-types >= 1.0 && < 2
@@ -38,7 +37,6 @@ Library
  Exposed-Modules:
    Data.UUID
    Data.UUID.Util
-   Data.UUID.V1
    Data.UUID.V3
    Data.UUID.V4
    Data.UUID.V5
-- 
2.1.4

