From 62cc9e3f70d8cea848d56efa198a68527fd07267 Mon Sep 17 00:00:00 2001
From: Joey Hess <joey@kitenet.net>
Date: Thu, 28 Feb 2013 23:40:19 -0400
Subject: [PATCH] avoid TH

---
 Yesod/Persist.hs       |    2 --
 yesod-persistent.cabal |    1 -
 2 files changed, 3 deletions(-)

diff --git a/Yesod/Persist.hs b/Yesod/Persist.hs
index 0646152..5130497 100644
--- a/Yesod/Persist.hs
+++ b/Yesod/Persist.hs
@@ -7,11 +7,9 @@ module Yesod.Persist
     , get404
     , getBy404
     , module Database.Persist
-    , module Database.Persist.TH
     ) where
 
 import Database.Persist
-import Database.Persist.TH
 import Control.Monad.Trans.Class (MonadTrans)
 
 import Yesod.Handler
diff --git a/yesod-persistent.cabal b/yesod-persistent.cabal
index 111c1b9..07f6e17 100644
--- a/yesod-persistent.cabal
+++ b/yesod-persistent.cabal
@@ -16,7 +16,6 @@ library
     build-depends:   base                      >= 4        && < 5
                    , yesod-core                >= 1.1      && < 1.2
                    , persistent                >= 1.0      && < 1.2
-                   , persistent-template       >= 1.0      && < 1.2
                    , transformers              >= 0.2.2    && < 0.4
     exposed-modules: Yesod.Persist
     ghc-options:     -Wall
-- 
1.7.10.4

