From 8bf7c428a42b984f63f435bb34f22743202ae449 Mon Sep 17 00:00:00 2001
From: foo <foo@bar>
Date: Sun, 22 Sep 2013 05:24:19 +0000
Subject: [PATCH] hacked up for Android

---
 Yesod.hs              |    2 --
 Yesod/Default/Util.hs |   17 -----------------
 2 files changed, 19 deletions(-)

diff --git a/Yesod.hs b/Yesod.hs
index b367144..3050bf5 100644
--- a/Yesod.hs
+++ b/Yesod.hs
@@ -5,9 +5,7 @@ module Yesod
     ( -- * Re-exports from yesod-core
       module Yesod.Core
     , module Yesod.Form
-    , module Yesod.Persist
     ) where
 
 import Yesod.Core
 import Yesod.Form
-import Yesod.Persist
diff --git a/Yesod/Default/Util.hs b/Yesod/Default/Util.hs
index a10358e..c5a4e58 100644
--- a/Yesod/Default/Util.hs
+++ b/Yesod/Default/Util.hs
@@ -8,7 +8,6 @@ module Yesod.Default.Util
     , widgetFileNoReload
     , widgetFileReload
     , TemplateLanguage (..)
-    , defaultTemplateLanguages
     , WidgetFileSettings
     , wfsLanguages
     , wfsHamletSettings
@@ -20,9 +19,6 @@ import Yesod.Core -- purposely using complete import so that Haddock will see ad
 import Control.Monad (when, unless)
 import System.Directory (doesFileExist, createDirectoryIfMissing)
 import Language.Haskell.TH.Syntax
-import Text.Lucius (luciusFile, luciusFileReload)
-import Text.Julius (juliusFile, juliusFileReload)
-import Text.Cassius (cassiusFile, cassiusFileReload)
 import Text.Hamlet (HamletSettings, defaultHamletSettings)
 import Data.Maybe (catMaybes)
 import Data.Default (Default (def))
@@ -69,24 +65,11 @@ data TemplateLanguage = TemplateLanguage
     , tlReload :: FilePath -> Q Exp
     }
 
-defaultTemplateLanguages :: HamletSettings -> [TemplateLanguage]
-defaultTemplateLanguages hset =
-    [ TemplateLanguage False "hamlet"  whamletFile' whamletFile'
-    , TemplateLanguage True  "cassius" cassiusFile  cassiusFileReload
-    , TemplateLanguage True  "julius"  juliusFile   juliusFileReload
-    , TemplateLanguage True  "lucius"  luciusFile   luciusFileReload
-    ]
-  where
-    whamletFile' = whamletFileWithSettings hset
-
 data WidgetFileSettings = WidgetFileSettings
     { wfsLanguages :: HamletSettings -> [TemplateLanguage]
     , wfsHamletSettings :: HamletSettings
     }
 
-instance Default WidgetFileSettings where
-    def = WidgetFileSettings defaultTemplateLanguages defaultHamletSettings
-
 widgetFileNoReload :: WidgetFileSettings -> FilePath -> Q Exp
 widgetFileNoReload wfs x = combine "widgetFileNoReload" x False $ wfsLanguages wfs $ wfsHamletSettings wfs
 
-- 
1.7.10.4

