From b7e01a2fded6575678db234e1f2de1f104f11376 Mon Sep 17 00:00:00 2001
From: Joey Hess <joey@kitenet.net>
Date: Mon, 15 Apr 2013 15:25:07 -0400
Subject: [PATCH 3/3] exports for TH splices

---
 Yesod/Widget.hs | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Yesod/Widget.hs b/Yesod/Widget.hs
index bf79150..01ae294 100644
--- a/Yesod/Widget.hs
+++ b/Yesod/Widget.hs
@@ -52,6 +52,9 @@ module Yesod.Widget
     , addScriptEither
       -- * Internal
     , unGWidget
+
+    -- used by TH code
+    , liftW
     ) where
 
 import Data.Monoid
-- 
1.8.2.rc3

