From a6fb84e676bc13a248baf9f617fc3e84ef902111 Mon Sep 17 00:00:00 2001
From: Joey Hess <joeyh@joeyh.name>
Date: Tue, 21 Nov 2017 13:10:35 -0400
Subject: [PATCH] disable use of accept4, for compatability with older systems

---
 Network/Socket.hsc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Network/Socket.hsc b/Network/Socket.hsc
index 6a1ce6a..ef4004e 100644
--- a/Network/Socket.hsc
+++ b/Network/Socket.hsc
@@ -535,7 +535,7 @@ accept sock@(MkSocket s family stype protocol status) = do
                 return new_sock
 #else
      with (fromIntegral sz) $ \ ptr_len -> do
-# ifdef HAVE_ACCEPT4
+# if 0
      new_sock <- throwSocketErrorIfMinus1RetryMayBlock "Network.Socket.accept"
                         (threadWaitRead (fromIntegral s))
                         (c_accept4 s sockaddr ptr_len (#const SOCK_NONBLOCK))
-- 
2.15.0

