diff --git a/AndroidManifest.xml b/AndroidManifest.xml
index b0e866a..1ab8515 100644
--- a/AndroidManifest.xml
+++ b/AndroidManifest.xml
@@ -7,6 +7,7 @@
     <uses-feature android:name="android.hardware.touchscreen" android:required="false" />
     <uses-permission android:name="android.permission.INTERNET" />
     <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
+    <uses-permission android:name="android.permission.WRITE_MEDIA_STORAGE" />
     <uses-permission android:name="android.permission.WAKE_LOCK" />
     <permission android:name="jackpal.androidterm.permission.RUN_SCRIPT"
                 android:label="@string/perm_run_script"
diff --git a/examples/widget/src/jackpal/androidterm/sample/telnet/TermActivity.java b/examples/widget/src/jackpal/androidterm/sample/telnet/TermActivity.java
index f6952f0..1a8df8f 100644
--- a/examples/widget/src/jackpal/androidterm/sample/telnet/TermActivity.java
+++ b/examples/widget/src/jackpal/androidterm/sample/telnet/TermActivity.java
@@ -166,7 +166,7 @@ public class TermActivity extends Activity
         /* ... create a process ... */
         String execPath = LaunchActivity.getDataDir(this) + "/bin/execpty";
         ProcessBuilder execBuild =
-                new ProcessBuilder(execPath, "/system/bin/sh", "-");
+                new ProcessBuilder(execPath, "/data/data/ga.androidterm/lib/lib.start.so", "");
         execBuild.redirectErrorStream(true);
         Process exec = null;
         try {
diff --git a/res/menu/main.xml b/res/menu/main.xml
index 064f833..fe5f3a3 100644
--- a/res/menu/main.xml
+++ b/res/menu/main.xml
@@ -16,6 +16,8 @@
 -->
 
 <menu xmlns:android="http://schemas.android.com/apk/res/android">
+    <item android:id="@+id/menu_send_email"
+        android:title="@string/send_email" />
     <item android:id="@+id/menu_new_window"
         android:title="@string/new_window"
         android:icon="@drawable/ic_menu_add" />
@@ -34,8 +36,6 @@
         android:icon="@drawable/ic_menu_preferences" />
     <item android:id="@+id/menu_reset"
         android:title="@string/reset" />
-    <item android:id="@+id/menu_send_email"
-        android:title="@string/send_email" />
     <item android:id="@+id/menu_toggle_wakelock"
         android:title="@string/enable_wakelock" />
     <item android:id="@+id/menu_toggle_wifilock"
diff --git a/res/values/defaults.xml b/res/values/defaults.xml
index 67287b2..9b7cfcd 100644
--- a/res/values/defaults.xml
+++ b/res/values/defaults.xml
@@ -13,10 +13,10 @@
    <string name="pref_fnkey_default">4</string>
    <string name="pref_ime_default">0</string>
    <bool name="pref_alt_sends_esc_default">false</bool>
-   <string name="pref_shell_default">/system/bin/sh -</string>
-   <string name="pref_initialcommand_default"></string>
+   <string name="pref_shell_default">/data/data/ga.androidterm/lib/lib.start.so</string>
+   <string name="pref_initialcommand_default">git annex webapp</string>
    <string name="pref_termtype_default">screen</string>
-   <bool name="pref_close_window_on_process_exit_default">true</bool>
+   <bool name="pref_close_window_on_process_exit_default">false</bool>
    <bool name="pref_verify_path_default">true</bool>
    <bool name="pref_do_path_extensions_default">true</bool>
    <bool name="pref_allow_prepend_path_default">true</bool>
diff --git a/res/values/strings.xml b/res/values/strings.xml
index f1464e9..b06ec9a 100644
--- a/res/values/strings.xml
+++ b/res/values/strings.xml
@@ -23,7 +23,7 @@
    <string name="prev_window">Prev window</string>
    <string name="next_window">Next window</string>
    <string name="reset">Reset term</string>
-   <string name="send_email">Email to</string>
+   <string name="send_email">Open WebApp</string>
    <string name="special_keys">Special keys</string>
    <string name="toggle_soft_keyboard">Toggle soft keyboard</string>
 
diff --git a/src/jackpal/androidterm/Term.java b/src/jackpal/androidterm/Term.java
index 8a3a4ac..af8d1ad 100644
--- a/src/jackpal/androidterm/Term.java
+++ b/src/jackpal/androidterm/Term.java
@@ -21,6 +21,9 @@ import java.text.Collator;
 import java.util.Arrays;
 import java.util.Locale;
 
+import java.io.FileReader;
+import java.io.BufferedReader;
+
 import android.app.Activity;
 import android.app.AlertDialog;
 import android.content.ActivityNotFoundException;
@@ -59,6 +62,11 @@ import android.view.inputmethod.InputMethodManager;
 import android.widget.TextView;
 import android.widget.Toast;
 
+import android.content.Intent;
+import android.net.Uri;
+import android.app.Activity;
+import android.content.Context;
+
 import jackpal.androidterm.emulatorview.ColorScheme;
 import jackpal.androidterm.emulatorview.EmulatorView;
 import jackpal.androidterm.emulatorview.TermSession;
@@ -911,31 +919,15 @@ public class Term extends Activity implements UpdateCallback {
     }
 
     private void doEmailTranscript() {
+	// Hack: repurposed to open the git-annex webapp
         TermSession session = getCurrentTermSession();
         if (session != null) {
-            // Don't really want to supply an address, but
-            // currently it's required, otherwise nobody
-            // wants to handle the intent.
-            String addr = "user@example.com";
-            Intent intent =
-                    new Intent(Intent.ACTION_SENDTO, Uri.parse("mailto:"
-                            + addr));
-
-            String subject = getString(R.string.email_transcript_subject);
-            String title = session.getTitle();
-            if (title != null) {
-                subject = subject + " - " + title;
-            }
-            intent.putExtra(Intent.EXTRA_SUBJECT, subject);
-            intent.putExtra(Intent.EXTRA_TEXT,
-                    session.getTranscriptText().trim());
             try {
-                startActivity(Intent.createChooser(intent,
-                        getString(R.string.email_transcript_chooser_title)));
-            } catch (ActivityNotFoundException e) {
-                Toast.makeText(this,
-                        R.string.email_transcript_no_email_activity_found,
-                        Toast.LENGTH_LONG).show();
+              BufferedReader buf = new BufferedReader(new FileReader("/sdcard/git-annex.home/.git-annex-url"));
+              String s = buf.readLine();
+              Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(s));
+              startActivity(intent);
+	    } catch (Exception e) {
             }
         }
     }
diff --git a/tools/build-debug b/tools/build-debug
index 1f15cd2..e611956 100755
--- a/tools/build-debug
+++ b/tools/build-debug
@@ -34,4 +34,4 @@ fi
 
 rm -rf `find . -name bin -o -name obj -prune`
 cd jni
-$ANDROID_NDK_ROOT/ndk-build && cd .. && ant debug
+$ANDROID_NDK_ROOT/ndk-build && cd ..
diff --git a/tools/update.sh b/tools/update.sh
index 57219c3..79b45ef 100755
--- a/tools/update.sh
+++ b/tools/update.sh
@@ -18,7 +18,7 @@ command -v "$ANDROID" >/dev/null 2>&1 || { echo >&2 "The $ANDROID tool is not fo
 
 # Make sure target-11 is installed
 
-$ANDROID update sdk -u -t android-11
+$ANDROID update sdk -u -t android-17
 
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 ATE_ROOT="$( cd $DIR/.. && pwd )"
@@ -31,5 +31,5 @@ for PROJECT_FILE in $PROJECT_FILES
 do
     PROJECT_DIR="$( dirname "$PROJECT_FILE" )"
     echo "Updating $PROJECT_FILE"
-    $ANDROID update project -p "$PROJECT_DIR" --target android-11
+    $ANDROID update project -p "$PROJECT_DIR" --target android-17
 done
