From ec2543b2156dc968530ad98e33d26f741a479e31 Mon Sep 17 00:00:00 2001
From: dummy <dummy@example.com>
Date: Tue, 14 Oct 2014 01:23:56 +0000
Subject: [PATCH] hack to build

---
 Database/Persist/Sql/Raw.hs |    6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/Database/Persist/Sql/Raw.hs b/Database/Persist/Sql/Raw.hs
index 319d75f..db3ee43 100644
--- a/Database/Persist/Sql/Raw.hs
+++ b/Database/Persist/Sql/Raw.hs
@@ -15,7 +15,7 @@ import Data.IORef (writeIORef, readIORef, newIORef)
 import Control.Exception (throwIO)
 import Control.Monad (when, liftM)
 import Data.Text (Text, pack)
-import Control.Monad.Logger (logDebugS, runLoggingT)
+-- import Control.Monad.Logger (logDebugS, runLoggingT)
 import Data.Int (Int64)
 import qualified Data.Text as T
 import Data.Conduit
@@ -39,8 +39,6 @@ rawQueryRes
 rawQueryRes sql vals = do
     conn <- ask
     let make = do
-            runLoggingT ($logDebugS (pack "SQL") $ pack $ show sql ++ " " ++ show vals)
-                (connLogFunc conn)
             getStmtConn conn sql
     return $ do
         stmt <- mkAcquire make stmtReset
@@ -52,8 +50,6 @@ rawExecute x y = liftM (const ()) $ rawExecuteCount x y
 rawExecuteCount :: MonadIO m => Text -> [PersistValue] -> ReaderT Connection m Int64
 rawExecuteCount sql vals = do
     conn <- ask
-    runLoggingT ($logDebugS (pack "SQL") $ pack $ show sql ++ " " ++ show vals)
-        (connLogFunc conn)
     stmt <- getStmt sql
     res <- liftIO $ stmtExecute stmt vals
     liftIO $ stmtReset stmt
-- 
1.7.10.4

