From 7e85a025349877565a70c375ef55508f215eaaf8 Mon Sep 17 00:00:00 2001
From: dummy <dummy@example.com>
Date: Wed, 21 May 2014 04:23:49 +0000
Subject: [PATCH] remove TH

---
 process-conduit.cabal | 1 -
 1 file changed, 1 deletion(-)

diff --git a/process-conduit.cabal b/process-conduit.cabal
index e6988e0..a2e03e0 100644
--- a/process-conduit.cabal
+++ b/process-conduit.cabal
@@ -24,7 +24,6 @@ source-repository head
 
 library
   exposed-modules:     Data.Conduit.Process
-                       System.Process.QQ
 
   build-depends:       base             == 4.*
                      , template-haskell >= 2.4
-- 
2.0.0.rc2

