From 85917e8b5da3c67c6ca0791fdad735ffb864ae3b Mon Sep 17 00:00:00 2001
From: dummy <dummy@example.com>
Date: Tue, 14 Oct 2014 02:50:19 +0000
Subject: [PATCH] not needed

---
 yesod-persistent.cabal |   10 ----------
 1 file changed, 10 deletions(-)

diff --git a/yesod-persistent.cabal b/yesod-persistent.cabal
index 2e5735d..438c76d 100644
--- a/yesod-persistent.cabal
+++ b/yesod-persistent.cabal
@@ -14,16 +14,6 @@ description:     Some helpers for using Persistent from Yesod.
 
 library
     build-depends:   base                      >= 4        && < 5
-                   , yesod-core                >= 1.4.0    && < 1.5
-                   , persistent                >= 2.1      && < 2.2
-                   , persistent-template       >= 2.1      && < 2.2
-                   , transformers              >= 0.2.2
-                   , blaze-builder
-                   , conduit
-                   , resourcet                 >= 0.4.5
-                   , resource-pool
-    exposed-modules: Yesod.Persist
-                     Yesod.Persist.Core
     ghc-options:     -Wall
 
 test-suite test
-- 
1.7.10.4

