From 4d8650bd806f50aa2538270f80fa93261c43d056 Mon Sep 17 00:00:00 2001
From: dummy <dummy@example.com>
Date: Fri, 3 Jul 2015 00:12:02 +0000
Subject: [PATCH] stub out

---
 yesod-persistent.cabal | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/yesod-persistent.cabal b/yesod-persistent.cabal
index c3bc1bf..1727dba 100644
--- a/yesod-persistent.cabal
+++ b/yesod-persistent.cabal
@@ -15,16 +15,6 @@ extra-source-files: README.md ChangeLog.md
 
 library
     build-depends:   base                      >= 4        && < 5
-                   , yesod-core                >= 1.4.0    && < 1.5
-                   , persistent                >= 2.1      && < 2.2
-                   , persistent-template       >= 2.1      && < 2.2
-                   , transformers              >= 0.2.2
-                   , blaze-builder
-                   , conduit
-                   , resourcet                 >= 0.4.5
-                   , resource-pool
-    exposed-modules: Yesod.Persist
-                     Yesod.Persist.Core
     ghc-options:     -Wall
 
 test-suite test
-- 
2.1.4

