From 583575461dc58b76c6c7e14d429f73182d49ef81 Mon Sep 17 00:00:00 2001
From: dummy <dummy@example.com>
Date: Tue, 10 Jun 2014 20:29:51 +0000
Subject: [PATCH] don't really build

---
 yesod-auth.cabal | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/yesod-auth.cabal b/yesod-auth.cabal
index 906c08b..b4bc841 100644
--- a/yesod-auth.cabal
+++ b/yesod-auth.cabal
@@ -65,17 +65,6 @@ library
                    , conduit-extra
                    , attoparsec-conduit
 
-    exposed-modules: Yesod.Auth
-                     Yesod.Auth.BrowserId
-                     Yesod.Auth.Dummy
-                     Yesod.Auth.Email
-                     Yesod.Auth.OpenId
-                     Yesod.Auth.Rpxnow
-                     Yesod.Auth.Message
-                     Yesod.Auth.GoogleEmail
-                     Yesod.Auth.GoogleEmail2
-    other-modules:   Yesod.Auth.Routes
-                     Yesod.PasswordStore
     ghc-options:     -Wall
 
 source-repository head
-- 
2.0.0

