From f6bfe8e01d8fe6d129ad3819070aa17934094a0a Mon Sep 17 00:00:00 2001
From: foo <foo@bar>
Date: Sun, 22 Sep 2013 06:24:09 +0000
Subject: [PATCH] export module referenced by TH splices

---
 yesod-routes.cabal |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/yesod-routes.cabal b/yesod-routes.cabal
index 0b245f2..a97582a 100644
--- a/yesod-routes.cabal
+++ b/yesod-routes.cabal
@@ -27,11 +27,11 @@ library
                      Yesod.Routes.Class
                      Yesod.Routes.Parse
                      Yesod.Routes.Overlap
+                     Yesod.Routes.TH.Types
     other-modules:   Yesod.Routes.TH.Dispatch
                      Yesod.Routes.TH.RenderRoute
                      Yesod.Routes.TH.ParseRoute
                      Yesod.Routes.TH.RouteAttrs
-                     Yesod.Routes.TH.Types
     ghc-options:     -Wall
 
 test-suite runtests
-- 
1.7.10.4

