From 9a8637661270d3c6a15e6aeca5e8983fe126bd27 Mon Sep 17 00:00:00 2001
From: dummy <dummy@example.com>
Date: Wed, 21 May 2014 05:27:36 +0000
Subject: [PATCH] don't really build

---
 yesod-auth.cabal | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/yesod-auth.cabal b/yesod-auth.cabal
index 0872581..723cde0 100644
--- a/yesod-auth.cabal
+++ b/yesod-auth.cabal
@@ -60,16 +60,6 @@ library
                    , byteable
                    , binary
 
-    exposed-modules: Yesod.Auth
-                     Yesod.Auth.BrowserId
-                     Yesod.Auth.Dummy
-                     Yesod.Auth.Email
-                     Yesod.Auth.OpenId
-                     Yesod.Auth.Rpxnow
-                     Yesod.Auth.Message
-                     Yesod.Auth.GoogleEmail
-    other-modules:   Yesod.Auth.Routes
-                     Yesod.PasswordStore
     ghc-options:     -Wall
 
 source-repository head
-- 
2.0.0.rc2

